src/components/Details.jsx

```diff
@@ -0,0 +1,62 @@
+import React from 'react';
+import { getUserData, saveUserData } from '../service';
+
+class Details extends React.Component {
+  state = {
+    petName: '',
+    roomName: '',
+    email: '',
+  }
+
+  async componentDidMount() {
+    await this.getDetails();
```

Comment:
I'm wondering if we should have the get details in the parent component Settings. it seems kind of a waste that it has to request it once per component even though it's for the same info.